### PR TITLE
[wip] Add external library (UTF8-CPP)

### PR DESCRIPTION
**Pull request by schnek**
- [x] todo: update encoding to world_db
- [ ] support declension Russian client
- [ ] support method void WorldSession::handleDeclinedPlayerNameOpcode { }

- [ ] concole (72c138b18783149c6d0c1f80896a5429287bbc55)

> [DEBUG] WorldSession::handleNameQueryOpcode Received CMSG_NAME_QUERY for: "names".
> [DETAIL] Player::SendInitialLogonPackets WORLD: Sent initial logon packets for "names".
> [DETAIL] WorldSession::FullLogin WORLD: Created new player for existing players "names".
[DEBUG] WorldSession::FullLogin Player "names" logged in.
[DETAIL] WorldSession::FullLogin WORLD: Created new player for existing players "names".
[DETAIL] Player::SendInitialLogonPackets WORLD: Sent initial logon packets for "names".
[DETAIL] Player::ApplyLevelInfo Player "names" set parameters to level 5
[DEBUG] WorldSession::handleNameQueryOpcode Received CMSG_NAME_QUERY for: "names".

### CODE REVIEW

**Comment by Zyres** [comment]
how is it going?

**Comment by schnek** [comment]
until now the library does not want to connect ;(

File: src/world/Server/Packets/Handlers/CharacterHandler.cpp [hunk 1]
```diff
@@ -1523,9 +1523,9 @@ void WorldSession::HandleDeclinedPlayerNameOpcode(WorldPacket& recv_data)
 
     recv_data >> guid;
 
-    std::string declinedname[6];
+    std::string declinedname[MAX_DECLINED_NAME_CASES];
 
-    for(int i = 0; i < 6; ++i)
+    for(int i = 0; i < MAX_DECLINED_NAME_CASES; ++i)
     {
```
Comment (armm77) [inline src/world/Server/Packets/Handlers/CharacterHandler.cpp]: and if you also change int for int8 ?

Comment (schnek) [reply src/world/Server/Packets/Handlers/CharacterHandler.cpp]: uint8_t in the future ..

**Comment by schnek** [comment]
update encoding to world_db https://github.com/AscEmu/OneDB/commit/700da485c3991784025dbda39a2c456a5c08d439
I will need help in PR 😌

**Pull request closed without merging**